packages/core/src/indexing/postgres/__tests__/postgres-index-api.test.ts

```diff
@@ -186,6 +221,51 @@ describe('indexStream', () => {
     const createdAt = new Date(raw.created_at)
     expect(closeDates(createdAt, createTime)).toBeTruthy()
   })
+
+  test('verify jsonb index creation and invocation while querying', async () => {
+    await indexApi.indexStream(STREAM_CONTENT_A)
+
+    // create index on jsonb content and disable seq scans
+    await dbConnection.raw(`SET enable_seqscan = off;`)
+    let result: Array<any> = await dbConnection.raw(
+      `CREATE INDEX idx_postgres_jsonb ON ${STREAM_ID_A}((stream_content->'settings'->'dark_mode'))`
+    )
+    expect(result.command).toEqual('CREATE')
+
+    // query indexed jsonb data
+    result = await dbConnection.raw(
+      `EXPLAIN SELECT *
+       FROM ${MODELS_TO_INDEX[0]}
+       WHERE stream_content->'settings'->'dark_mode' = ?;`,
+      true
+    )
+    expect(result.command).toEqual('EXPLAIN')
+    expect(result.rows.length).toBeGreaterThan(1)
+    expect(result.rows[0]['QUERY PLAN'].includes('idx_postgres_jsonb')).toBeTruthy()
+  })
+
+  test('query and filter jsonb stream content', async () => {
```

Comment:
does this test pass if you run it on its own, without running the previous test that creates the JSONB index?

Reply:
No, it would fail as it relies on the INDEX to be created because it checks for the `idx_postgres_jsonb` to be present and properly invoked when data is being queried.

Reply:
Right that was my concern.  In that case the index should be created in `beforeAll` or `beforeEach` setup of the test fixture.  Tests shouldn't have side affects that affect other tests.